src/state/dashboard/reducer.js

```diff
@@ -0,0 +1,179 @@
+// TODO: samples and experiments will most likely be moved into their own reducers in the future
+const initialState = {
+  stats: {},
+  samples: {},
+  samplesOverTime: [],
+  experiments: {},
+  experimentsOverTime: [],
+  jobs: {},
+  timeOptions: {
+    range: 'day',
+    timePoints: []
+  },
+  isLoading: false
+};
+
+const dashboardReducer = (state = initialState, action) => {
+  switch (action.type) {
+    case 'DASHBOARD_REQUEST_SUCCESS': {
+      const { stats, samples, experiments } = action.data;
+      return {
+        ...state,
+        stats,
+        samples,
+        experiments
+      };
+    }
+    case 'DASHBOARD_TIME_REQUESTS_SUCCESS': {
+      const { samplesOverTime, experimentsOverTime, jobs } = action.data;
+      return {
+        ...state,
+        samplesOverTime,
+        experimentsOverTime,
+        jobs,
+        isLoading: false
+      };
+    }
+    case 'DASHBOARD_TIME_OPTIONS_UPDATED': {
+      const { timeOptions } = action.data;
+      return {
+        ...state,
+        timeOptions,
+        isLoading: true
+      };
+    }
+    default: {
+      return state;
+    }
+  }
+};
+
+export default dashboardReducer;
+
+// chart selectors for creating chart data for individual charts on dashboard
+
+export function getTotalLengthofQueuesByType(state) {
+  const stats = state.dashboard.stats;
+  return Object.keys(stats).map(jobType => {
+    return {
+      name: jobType.split('_')[0],
+      value: stats[jobType].open + stats[jobType].pending
+    };
+  });
+}
+
+export function getJobsByStatus(state) {
+  const stats = state.dashboard.stats;
+  return Object.keys(stats).reduce((accum, jobType) => {
+    accum[jobType] = Object.keys(stats[jobType]).reduce(
+      (allStatuses, status) => {
+        if (status !== 'total' && status !== 'average_time') {
+          allStatuses.push({
+            name: status,
+            value: stats[jobType][status]
+          });
+        }
+        return allStatuses;
+      },
+      []
+    );
+    return accum;
+  }, {});
+}
+
+function convertSecToMinHours(sec) {
+  const hours = Math.floor(sec / 3600),
+    minutes = Math.floor((sec % 3600) / 60);
+  if (isNaN(hours) || isNaN(minutes)) {
+    return `N/A`;
+  } else {
+    return `${hours} hr ${minutes} min`;
+  }
+}
+
+export function getAllEstimatedTimeTilCompletion(state, jobType) {
+  const stats = state.dashboard.stats;
+
+  if (!Object.keys(stats).length) return {};
+
+  return Object.keys(stats).reduce((allEstimatedTimes, jobType) => {
+    const estimateSec =
+      (stats[jobType].open + stats[jobType].pending) *
+      parseInt(stats[jobType].average_time, 10);
```

Comment:
Wouldn't `Math.round()` make more sense here?

Reply:
`average_time` seems to get returned as a string, so I need to make it an integer.

Reply:
What I meant was that using `Math.round()` will be a little more accurate. But since this is just an estimate, it doesn't really matter that much whether a value like `9.8` is converted to 9 or 10.

Reply:
ahh yes, probably `parseFloat()` would actually work best here, so it doesn't remove the decimals